library/manageiq_provider.py

```diff
@@ -465,11 +481,13 @@ def main():
 
     if state == 'present':
         if provider_type in ("openshift-enterprise", "openshift-origin"):
-            endpoints = [manageiq.generate_openshift_endpoint('default', 'bearer', hostname, port, token)]
+            endpoints = [manageiq.generate_auth_key_config('default', 'bearer', hostname, port, token)]
             if module.params['metrics']:
-                endpoints.append(manageiq.generate_openshift_endpoint('hawkular', 'hawkular', h_hostname, h_port, token))
+                endpoints.append(manageiq.generate_auth_key_config('hawkular', 'hawkular', h_hostname, h_port, token))
         elif provider_type == "amazon":
-            endpoints = [manageiq.generate_amazon_endpoint('default', 'default', access_key_id, secret_access_key)]
+            endpoints = [manageiq.generate_amazon_config('default', 'default', access_key_id, secret_access_key)]
+        elif provider_type == "hawkular-datawarehouse":
+            endpoints = [manageiq.generate_auth_key_config('default', 'default', hostname, port, token)]
```

Comment:
I think we should include `"hawkular-datawarehouse"` in the first providers `if` block `("openshift-enterprise", "openshift-origin")` and add `PROVIDER_AUTH_TYPES` dict in `ManageIQ` class, to include these constants `('default'` for hawkular, `'bearer'` for the openshift providers). 
Perhaps also do that for the roles.

Not a blocker for this change, can also be part of a future refactor